src/internal/observable/range.ts

```diff
@@ -1,3 +1,80 @@
-import {  RangeObservable  } from './RangeObservable';
+import { IScheduler } from '../Scheduler';
+import { Observable } from '../Observable';
+import { Action } from '../scheduler/Action';
 
-export const range = RangeObservable.create;
\ No newline at end of file
+/**
+ * Creates an Observable that emits a sequence of numbers within a specified
+ * range.
+ *
+ * <span class="informal">Emits a sequence of numbers in a range.</span>
+ *
+ * <img src="./img/range.png" width="100%">
+ *
+ * `range` operator emits a range of sequential integers, in order, where you
+ * select the `start` of the range and its `length`. By default, uses no
+ * IScheduler and just delivers the notifications synchronously, but may use
+ * an optional IScheduler to regulate those deliveries.
+ *
+ * @example <caption>Emits the numbers 1 to 10</caption>
+ * var numbers = Rx.Observable.range(1, 10);
+ * numbers.subscribe(x => console.log(x));
+ *
+ * @see {@link timer}
+ * @see {@link interval}
+ *
+ * @param {number} [start=0] The value of the first integer in the sequence.
+ * @param {number} [count=0] The number of sequential integers to generate.
+ * @param {Scheduler} [scheduler] A {@link IScheduler} to use for scheduling
+ * the emissions of the notifications.
+ * @return {Observable} An Observable of numbers that emits a finite range of
+ * sequential integers.
+ * @static true
+ * @name range
+ * @owner Observable
+ */
+export function range(start: number = 0,
+                      count: number = 0,
+                      scheduler?: IScheduler): Observable<number> {
+  return new Observable<number>(subscriber => {
+    let index = 0;
+
+    if (scheduler) {
+      return scheduler.schedule(dispatch, 0, {
+        index, count, start, subscriber
+      });
+    } else {
+      do {
+        if (index++ >= count) {
+          subscriber.complete();
+          break;
+        }
+        subscriber.next(start++);
+        if (subscriber.closed) {
+          break;
+        }
+      } while (true);
+    }
+
+    return undefined;
```

Comment:
when this can execution path can be reached?

Reply:
This was added to make TS happy, it was warning that not all code paths returned a value.

Reply:
sounds weird (that TS doesn't check branching correctly) 🤔 , but since it's non excutable anyway.. I'd suggest to change `throw new Error()` (and if anyone actually hit those, give user T-shirts for prize.)

Reply:
But you *can* hit it.

Reply:
Uhhh yes you are right. I got confused above branch return early for all cases. Good as is then.